### DISCUSSION TITLE
G-Codes for Bed Level visualizer

### DISCUSSION THREAD

**Discussion by triton-nereide** [jneilliii/OctoPrint-BedLevelVisualizer#622] [Q&A] [answered]
I have a question about BedevelVisualizer: I own a Prusa MK3+ - do I need to insert the code for bilinear bedleveling AND the few lines for Prusa Mk3, or only the lines for Prusa MK3+? Or maybee any other combination?

regards, C

**Comment by jneilliii** [comment, marked as answer]
you should only need the commands necessary to report the mesh if you just want to graph what's current, otherwise you need to include the commands necessary to do the probing process as well. At this point, I'm unsure if the PrusaFirmware is using G29 like other Marlin based firmware or if it's still using the G80/G81 commands.

**Comment by HonzaJaros** [comment]
hi, 

what would be the g-code for 3.14.0 mk3s+? I am getting a lot of errors when the probing is done.
printer says calibration completed and then goes to restart


this is what octoprint shows:
```
Recv: echo:busy: processing
Recv: T:25.5 /0.0 B:26.3 /0.0 T0:25.5 /0.0 @:0 B@:0 P:0.0 A:34.3
Recv: echo:busy: processing
Recv: T:25.3 /0.0 B:26.1 /0.0 T0:25.3 /0.0 @:0 B@:0 P:0.0 A:34.4
Recv: echo:busy: processing
Recv: T:25.6 /0.0 B:26.2 /0.0 T0:25.6 /0.0 @:0 B@:0 P:0.0 A:34.3
Recv: echo:busy: processing
Recv: T:25.4 /0.0 B:26.2 /0.0 T0:25.4 /0.0 @:0 B@:0 P:0.0 A:34.3
Recv: echo:busy: processing
Recv: T:25.2 /0.0 B:26.4 /0.0 T0:25.2 /0.0 @:0 B@:0 P:0.0 A:34.1
Recv: echo:busy: processing
Recv: T:25.2 /0.0 B:26.1 /0.0 T0:25.2 /0.0 @:0 B@:0 P:0.0 A:34.4
Recv: echo:busy: processing
Recv: T:25.5 /0.0 B:26.2 /0.0 T0:25.5 /0.0 @:0 B@:0 P:0.0 A:34.2
Recv: echo:busy: processing
Recv: T:25.2 /0.0 B:26.1 /0.0 T0:25.2 /0.0 @:0 B@:0 P:0.0 A:34.0
Recv: echo:busy: processing
Recv: LCD status changed
Recv: Error:Heaters switched off. MINTEMP BED triggered!
Changing monitoring state from "Operational" to "Error"
```

**Reply by jneilliii** [reply]
what are you using currently? that just shows me what is happening and it doesn't appear to be probing at all. I assume it's going to be one of the Prusa variants listed [here](https://github.com/jneilliii/OctoPrint-BedLevelVisualizer/blob/master/wiki/gcode-examples.md#prusa-firmware-older-mk3-and-below).

**Reply by jneilliii** [reply]
Looking at the release notes of 3.14.0 firmware it appears there was some reworking/back-porting of the PrusaBuddy Firmware MBL methods. So there could be some changes necessary, but it appears that G81 should still report the mesh data. You just have to figure out the command to run `Z calibration`, which I assume is still G80. 

https://github.com/prusa3d/Prusa-Firmware/releases/tag/v3.14.0

**Reply by HonzaJaros** [reply]
yes
```
;
; MBL Prep
;
G80			; run mesh bed leveling routine.
@BEDLEVELVISUALIZER	; instruct plugin to start recording responses from printer.
G81			; report mesh bed leveling status.
```
and flipped the Y

**Reply by HonzaJaros** [reply]
it does what it supposed to, probs the bed, but at the end goes to the restart with minitemp error :(

**Reply by jneilliii** [reply]
I'm not going to be much help then. The firmware is doing weird stuff it sounds like to me and never completing its internal process, hence the continued processing commands and have no idea what would cause a min temp error outside a broken thermistor. When that error is sent to OctoPrint though it is causing OctoPrint to disconnect and probably send M112 command.
